inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImpl.java

```diff
@@ -312,15 +304,57 @@ public void setFeatureValue(AnnotationFeature aFeature, Object aPredictedValue,
     @Override
     public RecommenderContext getContext(User aUser, Recommender aRecommender)
     {
-        RecommendationUserState recommendationUserState = getState(aUser.getUsername());
-        RecommenderContext context = recommendationUserState.getContext(aRecommender);
-        if (context == null) {
-            context = new RecommenderContext();
-            recommendationUserState.putContext(aRecommender, context);
+        RecommendationUserState state = getState(aUser.getUsername(), aRecommender.getProject());
+        synchronized (state) {
+            return state.getContext(aRecommender);
         }
-        return context;
     }
 
+    private static class RecommentationStateKey
+    {
+        private final String user;
+        private final long projectId;
+        
+        public RecommentationStateKey(String aUser, long aProjectId)
+        {
+            super();
```

Comment:
Why do you need super?

Reply:
Some people think (including Eclipse) that it is good style to always call super or this in constructors to make the initialization hierarchy explicit. Removed it (as I usually always do).